marketplaceItems/avatarCustomization101/appResources/appData/avatarCustomization101.js

```diff
@@ -0,0 +1,937 @@
+
+        dynamicData[STRING_MATERIAL].selectedMaterial = materialName;
+
+        switch (materialName){
+            case STRING_DEFAULT_MAT:
+
+                setMaterialPropertiesToDefaults();
+                dynamicData[STRING_MATERIAL].selectedTypeIndex = 0; // "Select one"
+
+                if (materialID) {
+                    Entities.deleteEntity(materialID);
+                    materialID = null;
+                }
+
+                break;
+            case STRING_GLASS_MAT:
+                // updateMaterial materialName, isNamed, isPBR
+                updateMaterial(MATERIAL_GLASS, true, true);
+                break;
+            case STRING_CHAIN_MAT:
+                updateMaterial(MATERIAL_CHAINMAIL, true, true);
+                break;
+            case STRING_DISCO_MAT:
+                updateMaterial(MATERIAL_DISCO, true, true);
+                break;
+            case STRING_RED_MAT:
+                updateMaterial(MATERIAL_RED, true, false);
+                break;
+            case STRING_TEXTURE_MAT:
+                updateMaterial(MATERIAL_TEXTURE, true, false);
+                break;
+        }
+    }
+
+    // #endregion MATERIAL
+
+    // #region BLENDSHAPES
+
+    var TRANSITION_TIME_SECONDS = 0.25;
+    var STRING_AWE = "awe";
+    var STRING_ANGRY = "angry";
+    var STRING_DEFAULT = "default";
+    var STRING_LAUGH = "laugh";
+    var BLENDSHAPES_DEFAULT = BLENDSHAPE_DATA.defaults;
+    var BLENDSHAPES_AWE = BLENDSHAPE_DATA.awe;
+    var BLENDSHAPES_LAUGH = BLENDSHAPE_DATA.laugh;
+    var BLENDSHAPES_ANGRY = BLENDSHAPE_DATA.angry;
+
+    var lastEmotionUsed = BLENDSHAPES_DEFAULT; // values associated with the last emotion lerping
+    var isChangingEmotion = false; // transitioning
+    var changingEmotionPercentage = 0.0; // with transitioning
+    var isBlendshapeIntervalEnabled = false;
+
+    // used to mix between blendshape expressions
+    function mixBlendshapesInterval (deltaTime) {
+        if (!isChangingEmotion) {
+            return;
+        }
+
+        var blendshapeDynamicData = dynamicData[STRING_BLENDSHAPES];
+
+        changingEmotionPercentage += deltaTime / TRANSITION_TIME_SECONDS;
+        if (changingEmotionPercentage >= 1.0) {
+            changingEmotionPercentage = 1.0;
+            isChangingEmotion = false;
+        }
+        for (var blendshape in blendshapeDynamicData.updatedProperties) {
+            MyAvatar.setBlendshape(blendshape,
+                mixValue(lastEmotionUsed[blendshape], blendshapeDynamicData.updatedProperties[blendshape], changingEmotionPercentage));
+        }
+
+        // Helper function
+        function mixValue(valueA, valueB, percentage) {
+            return valueA + ((valueB - valueA) * percentage);
+        }
+    
+    }
+
+    function startBlendshapeInterval () {
+
+        if (!isBlendshapeIntervalEnabled) {
+            isBlendshapeIntervalEnabled = true;
+            Script.update.connect(mixBlendshapesInterval);
```

Comment:
if the script exits before this signal can be disconnected, you'll then have an update loop be connected erroneously. please add protection against that.

additionally, please don't use `Script.update.connect` unless it's absolutely necessary. having a script update something as fast as is absolutely possible is usually not necessary and adds to system load. do you think it's necessary here?